rsciio/tests/test_edax.py

```diff
@@ -29,9 +29,8 @@
 hs = pytest.importorskip("hyperspy.api", reason="hyperspy not installed")
 requests = pytest.importorskip("requests", reason="requests not installed")
 
-
-MY_PATH = os.path.dirname(__file__)
-ZIPF = os.path.join(MY_PATH, "edax_files.zip")
+TEST_DATA_PATH = Path(__file__).parent / "data" / "edax"
+ZIPF = TEST_DATA_PATH / "edax_files.zip"
 TMP_DIR = tempfile.TemporaryDirectory()
```

Comment:
use `tmp_path` here as for other plugins?

Reply:
maybe it is good chance to get rid of zip and other packed data?
Why:
1) bandwitch saving is redundant, as git compresses data behind anyway.
2) size is moot point, if this is going to be downloaded on demand
3) updating with new files, requires huge binary differences, getting rid of zip/gz collections, only new file will be added as difference to the git history.
4) finally, these tmp_path would be not needed (albeit if tests would be rewritten to use rsciio to read the data explicitly, and retrieved list/dictionary fed to hyperspy to generate signal for testing, for some formats loading could be straight from zip file, without any tmp_folders)

Reply:
Yes, we should try to remove of the zip files.

Just to avoid misunderstand, the expected usage is to download the files on demand when running from a non-development installation. For developer, there will be no change in workflow (well, you can remove the data folder and it will be downloaded again).
The main aim is to be able to run the test suite from the conda-forge feedstock or the hyperspy-bundle.

> 4. finally, these tmp_path would be not needed (albeit if tests would be rewritten to use rsciio to read the data explicitly, and retrieved list/dictionary fed to hyperspy to generate signal for testing, for some formats loading could be straight from zip file, without any tmp_folders)

I am not sure about that. What would be the benefit? For development purpose I usually prefer to open the file directly, sometimes using different approach, etc.

Reply:
if zip files are get rid (replaced with unpacked content), then no tmp_path is needed as all files can be just opened directly (using Hyperspy `load`). That is prefered way.

Sorry for convoluted message. What I had meant some of readers (undocumented behavior) are able to open not only physical files, but can accept the opened file or other `BytesIO` like opened object, i.e. file from within zip archive - no tmp folder then is needed, as byte-stream is read and parsed directly seamlessly from zip. I know that that works with tiffs. Ofc for test files that feature rather not good due to 1,2,3 reasons given before. However in the future we should consider documenting this behavior as gz, zip are one of most common available well documented, tested, widely supported containers. I.e. it is nice to be able to keep whole SEM session for one sample within single file.

Reply:
Ignoring the zip file business, for the scenario where `tmp_path` are used for testing saving file, we should keep this pattern, because we don't need to worry about removing them, user having write access to the folder, etc. What do you think?

This make me think that to run the test suite, the user will need write access to the tests folder, which is not guaranty in multi user installation. I have updated the summary of this PR.

Reply:
> use `tmp_path` here as for other plugins?

Same as above.

Reply:
of course for saving tests the situation is different, and rather it can't be evaded anyhow.